packages/core/src/agent/Agent.ts

```diff
@@ -41,7 +41,7 @@ interface AgentOptions<AgentModules extends AgentModulesInput> {
 // Any makes sure you can use Agent as a type without always needing to specify the exact generics for the agent
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export class Agent<AgentModules extends AgentModulesInput = any> extends BaseAgent<AgentModules> {
-  public messageSubscription: Subscription
+  public messageSubscription?: Subscription
```

Comment:
let's make it private

```suggestion
  private messageSubscription?: Subscription
```